src/waveform/waveform.h

```diff
@@ -31,6 +31,13 @@ union WaveformData {
 
 class Waveform {
   public:
+    enum class SaveState {
+        Virgin = 0,
+        Changing, 
```

Comment:
Mmm... ok, I think this is definitely better:
    enum class SaveState {
        NotSaved = 0,
        SavePending, 
        Saved
    };